### PR TITLE
Fixing intermittently failing killAndHealNode

### PR DESCRIPTION
**Pull request by zalokhan**
## Overview

Description:
Fixing intermittently failing killAndHealNode IT.
Reducing waiting times on unit tests.

Why should this be merged: 
Fixes intermittent failures.

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

**Comment by rogermichoud** [comment]
Sometimes I had issues running tests on my machine because timeouts were too aggressive. Am I the only one experiencing this? Also, we might start to see some timeout exceptions as well in our CI. This is also especially relevant for IT tests.

**Comment by Maithem** [comment]
ClusterReconfigIT doesn't actually pass on my machine because the timeout is too short.

**Comment by zalokhan** [comment]
I see. I only changed the retry timeouts. Probably I made them too aggressive.
I'll remove the timeout changes and keep the test fixes.

File: test/src/test/java/org/corfudb/integration/ClusterReconfigIT.java [hunk 1]
```diff
@@ -287,7 +287,8 @@ public void restartTest() throws Exception {
 
         restartServer(corfuRuntime, DEFAULT_ENDPOINT);
 
-        assertThat(corfuRuntime.getLayoutView().getLayout().getEpoch()).isEqualTo(l.getEpoch() + 1);
+        assertThat(corfuRuntime.getLayoutView().getLayout().getEpoch())
+                .isGreaterThanOrEqualTo(l.getEpoch() + 1);
```
Comment (Maithem) [inline test/src/test/java/org/corfudb/integration/ClusterReconfigIT.java]: This doesn't match the comment. Why do you expect the epoch to be greater than or equal, is it because recovery can fail and the epoch can increase multiple times?

File: test/src/test/java/org/corfudb/integration/ClusterReconfigIT.java [hunk 2]
```diff
@@ -466,8 +467,6 @@ public void killAndHealNode() throws Exception {
                 .open();
         final String data = createStringOfSize(1_000);
         Random r = getRandomNumberGenerator();
-        final AtomicBoolean moreDataToBeWritten = new AtomicBoolean(true);
```
Comment (Maithem) [inline test/src/test/java/org/corfudb/integration/ClusterReconfigIT.java]: Why did you remove this?

File: test/src/test/java/org/corfudb/integration/ClusterReconfigIT.java [hunk 3]
```diff
@@ -515,16 +514,12 @@ public void killAndHealNode() throws Exception {
         refreshedLayout = runtime.getLayoutView().getLayout();
         final long epochAfterHealingNode = 3L;
         // Waiting node to be healed and added back to the layout.
-        while (refreshedLayout.getEpoch() != epochAfterHealingNode) {
+        while (refreshedLayout.getEpoch() < epochAfterHealingNode) {
```
Comment (Maithem) [inline test/src/test/java/org/corfudb/integration/ClusterReconfigIT.java]: It seems like you have these while loops in many places. I dont think this is a good idea, probably need to bound the retries.

File: test/src/test/java/org/corfudb/integration/ClusterReconfigIT.java (hunk 1 above)
Comment (zalokhan) [reply test/src/test/java/org/corfudb/integration/ClusterReconfigIT.java]: Yes that's the reason. I'll update the comment to reflect this.

File: test/src/test/java/org/corfudb/integration/ClusterReconfigIT.java (hunk 2 above)
Comment (zalokhan) [reply test/src/test/java/org/corfudb/integration/ClusterReconfigIT.java]: This is creating a daemon writer thread initially introduced to check if we can write after killing or adding. I anyways have a write after these 2 events (in the main test thread) and fail if these writes fail.
So I feel this extra thread is not required now.

File: test/src/test/java/org/corfudb/integration/ClusterReconfigIT.java (hunk 3 above)
Comment (zalokhan) [reply test/src/test/java/org/corfudb/integration/ClusterReconfigIT.java]: Done.

**Comment by Maithem** [comment]
@annym is the injection-framework CI working, should we wait for it?

**Comment by annym** [comment]
@Maithem yes it is working, I restarted them and this one is already complete.

**Comment by Maithem** [comment]
@annym thanks, i see it now.
